app/helpers/admin/cancel_routes_helper.rb

```diff
@@ -1,24 +1,10 @@
 module Admin::CancelRoutesHelper
-  include ActionDispatch::Routing::PolymorphicRoutes
   include Admin::EditionRoutesHelper
-
   def admin_cancel_path(object)
-    if object.new_record?
-      case object
-      when CorporateInformationPage
-        polymorphic_path([:admin, object.owning_organisation, CorporateInformationPage])
-      when Edition
-        admin_editions_path
-      else
-        polymorphic_path([:admin, object.class])
-      end
+    if object.is_a? Edition
+      object.new_record? ? admin_editions_path : admin_edition_path(object)
```

Comment:
hmmm i'm just wondering if we can't move "cancel_path" to the model... (might be overloading the model a bit, maybe can have a discussion). so Edition always has a "cancel_path", and corporate information page can override that, which means we don't have the change the form view either, just for the cancel link

Reply:
I would actually go the other way and just call the correct method from the template, creating a new template if necessary for each usage. I didn't do that because I considered it outside of the scope of the PR, but happy to add a follow up.